flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/source/log/LogKafkaPartitionSplitState.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.flink.read.source.log;
+
+import org.apache.flink.connector.kafka.source.split.KafkaPartitionSplitState;
+import org.apache.flink.table.data.RowData;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+public class LogKafkaPartitionSplitState extends KafkaPartitionSplitState {
+
+    /**
+     * Denote reader is in retracting read mode.
+     * In this mode, data would be read in reverse order and opposite RowKind.
+     */
+    private boolean retracting;
```

Comment:
Could you add annotations for the following params? Helping devs understand them.